app/modules/enduser_setup.c

```diff
@@ -1272,47 +1398,69 @@ static int enduser_setup_manual (lua_State *L)
 
 static int enduser_setup_start(lua_State *L)
 {
-  ENDUSER_SETUP_DEBUG(L, "enduser_setup_start");
+  ENDUSER_SETUP_DEBUG("enduser_setup_start");
 
   if (!do_station_cfg_handle)
+  {
     do_station_cfg_handle = task_get_id(do_station_cfg);
+  }
+  if (!do_disconnect_tcp_handle)
+  {
+    do_disconnect_tcp_handle = task_get_id(do_disconnect_tcp);
+  }
 
   if(enduser_setup_init(L))
+  {
     goto failed;
+  }
 
+  enduser_setup_check_station_start();
   if (!manual)
   {
-    enduser_setup_check_station_start();
     enduser_setup_ap_start();
   }
 
   if(enduser_setup_dns_start())
+  {
     goto failed;
+  }
 
   if(enduser_setup_http_start())
+  {
     goto failed;
+  }
 
   goto out;
 
 failed:
-  enduser_setup_stop(L);
+  enduser_setup_stop();
 out:
   return 0;
 }
 
 
-static int enduser_setup_stop(lua_State* L)
+/**
+ * A wrapper needed for type-reasons strictness reasons.
+ */
+static void enduser_setup_stop_callback(void *ptr)
+{
+  enduser_setup_stop();
+}
+
+
+static void enduser_setup_stop()
 {
+  ENDUSER_SETUP_DEBUG("enduser_setup_stop");
+
   if (!manual)
   {
-    enduser_setup_check_station_stop();
     enduser_setup_ap_stop();
   }
   enduser_setup_dns_stop();
   enduser_setup_http_stop();
   enduser_setup_free();
 
-  return 0;
+  return;
```

Comment:
...and this in particular is likely to cause grief, as Lua uses the return value (here, undefined) to retrieve N number of return values from the stack.